pullrequest.rst

```diff
@@ -346,7 +346,7 @@ Now you want to
 <https://help.github.com/articles/creating-a-pull-request-from-a-fork/>`_.
 If this is a pull request in response to a pre-existing issue on the
 `issue tracker`_, please make sure to reference the issue number using
-``gh-NNNN`` in the pull request title or message.
+``gh-NNNNN: `` in the pull request title.
```

Comment:
Currently GitHub ignores the `gh-NNNNN: ` in the title, so a `gh-NNNNN` in the message helps adding an event in the issue timeline.  We should recommend both (at least until GitHub fixes title mentions:
```suggestion
``gh-NNNNN: `` in the pull request title and `gh-NNNNN` in the issue body.
```

Reply:
Since we have `gh-NNNNN` in title, how about just `#NNNNN` in body?

GitHub doesn't contain pull request body in squash commit message by default.
So `#NNNNN` won't be contained in the commit message.

If core developer copy&paste the pull request message to squash commit message, they would just trim the `#NNNNN`. They don't need to rewrite it to `gh-NNNNN` form because it is included in the first line of the commit message already.

Reply:
> Since we have gh-NNNNN in title, how about just #NNNNN in body?

That works too.  The goal of adding it in the body is just to reference the issue and create an event there, and both `gh-NNNNN` and `#NNNNN` work.